### PR TITLE
Remove old replay route

### PR DESCRIPTION
**Pull request by kmadsen**
## Description

Resolves: https://github.com/mapbox/mapbox-navigation-android/issues/3013

Delete ReplayRouteLocationEngine and migrate everything that uses it.

Status: ~It's compiling and it's runnable. But it doesn't work, so it's a draft. Opening because this will help verify if the new replay api changes feel good.~
It's all working, but now thinking about how to clean up the code

- [ ] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

## Migration status
This list includes all examples and usages that need to be migrated

- [x] BasicNavigationActivity https://github.com/mapbox/mapbox-navigation-android/pull/3046
- [x] DebugMapboxNavigationKt  https://github.com/mapbox/mapbox-navigation-android/pull/3046
- [x] FeedbackButtonActivity (in this change)
- [x] GuidanceViewActivity (in this change)
- [x] InstructionViewActivity (in this change)
- [x] SimpleMapboxNavigationKt (in this change)
- [x] SummaryBottomSheetActivity (in this change)
- [x] CustomUIComponentStyleActivity (in this change)
- [x] ReplayActivity
- [x] ReplayWaypointsActivity

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/MapboxReplayer.kt [hunk 1]
```diff
@@ -10,7 +10,8 @@ import com.mapbox.navigation.core.replay.history.ReplayEventSimulator
 import com.mapbox.navigation.core.replay.history.ReplayEventUpdateLocation
 import com.mapbox.navigation.core.replay.history.ReplayEvents
 import com.mapbox.navigation.core.replay.history.ReplayEventsObserver
-import com.mapbox.navigation.core.replay.route2.ReplayRouteMapper
+import com.mapbox.navigation.core.replay.route.ReplayRouteMapper
+import java.lang.Exception
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/MapboxReplayer.kt:14]:
```suggestion
```

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java [hunk 2]
```diff
@@ -207,8 +211,14 @@ void initialize(NavigationViewOptions options) {
       initializeVoiceInstructionCache();
       initializeNavigationSpeechPlayer(options);
       initializeMapOfflineManager(options);
+
+      if (navigationViewOptions.shouldSimulateRoute()) {
+        final Point lastLocation = getOriginOfRoute(options.directionsRoute());
+        ReplayEventBase replayEventOrigin = ReplayRouteMapper.mapToUpdateLocation(0.0, lastLocation);
+        mapboxReplayer.pushEvents(Collections.singletonList(replayEventOrigin));
+        mapboxReplayer.play();
+      }
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java:220]: Should this be moved into `initializeLocationEngineFrom` 👇?

**Comment by abhishek1508** [comment]
Yes you are correct. @Guardiola31337 
Question for you @kmadsen 
Apart from the example activities, anything that needs to be changed in the UI SDK as well in terms of ReplayEngine?

**Comment by Guardiola31337** [comment]
@abhishek1508 
> Apart from the example activities, anything that needs to be changed in the UI SDK as well in terms of ReplayEngine?

Changes were added 🚀 That's why I asked you for double checking 😅 - initial 👀 was looking good as mentioned.

**Comment by abhishek1508** [comment]
Ah okay. I just looked at Kyle's comment. Will look at the PR 😀

**Pull request merged**